test/e2e/operator_test.go

```diff
@@ -671,3 +671,92 @@ func TestHostNetworkEndpointPublishingStrategy(t *testing.T) {
 		t.Fatalf("WARNING: cloud resources may have been leaked! failed to delete ingresscontroller %s: %v", name, err)
 	}
 }
+
+// TestInternalLoadBalancer creates an ingresscontroller with the
+// "LoadBalancerService" endpoint publishing strategy type with scope set to
+// "Internal" and verifies that the operator creates a load balancer and that
+// the load balancer has a private IP address.
+func TestInternalLoadBalancer(t *testing.T) {
+	cl, ns, err := getClient()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	infraConfig := &configv1.Infrastructure{}
+	err = cl.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig)
+	if err != nil {
+		t.Fatalf("failed to get infrastructure config: %v", err)
+	}
+
+	annotation := map[string]string{}
+	switch infraConfig.Status.Platform {
+	case configv1.AWSPlatformType:
+		annotation["service.beta.kubernetes.io/aws-load-balancer-proxy-protocol"] = "*"
+	case configv1.AzurePlatformType:
+		annotation["service.beta.kubernetes.io/azure-load-balancer-internal"] = "true"
+	case configv1.BareMetalPlatformType:
+		t.Skip("test skipped on bare metal")
+	case configv1.GCPPlatformType:
+		annotation["cloud.google.com/load-balancer-type"] = "Internal"
+	case configv1.LibvirtPlatformType:
+		t.Skip("test skipped on libvirt")
+	case configv1.OpenStackPlatformType:
+		annotation["service.beta.kubernetes.io/openstack-internal-load-balancer"] = "true"
+	case configv1.NonePlatformType:
+		t.Skip("test skipped on \"none\" platform type")
+	case configv1.VSpherePlatformType:
+		t.Skip("test skipped on vSphere")
+	}
```

Comment:
TODO: Seems like it could be condensed to a lookup from internalLBAnnotations (if that were exposed by the package)